apps/wallet-mobile/src/yoroi-wallets/hooks/index.ts

```diff
@@ -985,34 +986,51 @@ export const useNativeAssetImage = ({
   width,
   height,
   mediaType = 'image/webp',
-  resizeMode = 'cover',
+  contentFit = 'cover',
   kind = 'metadata',
+  responseType = 'binary',
 }: NativeAssetImageRequest) => {
   const network = networkId === 300 ? 'preprod' : 'mainnet'
   const pWidth = PixelRatio.getPixelSizeForLayoutSize(Number(width))
   const pHeight = PixelRatio.getPixelSizeForLayoutSize(Number(height))
-  const isMediaTypeSupported = supportedTypes.includes(mediaType.toLocaleLowerCase())
+  const lcMediaType = mediaType.toLocaleLowerCase()
+  const isMediaTypeSupported = supportedTypes.includes(lcMediaType)
+  const needsGif = lcMediaType === 'image/gif' && Platform.OS === 'ios'
+  const mimeType = needsGif ? 'image/gif' : 'image/webp'
+  const headers = useMemo(
+    () => ({
+      Accept: responseType === 'binary' ? mimeType : 'text/plain',
+      'X-Encoded-Mimetype': mimeType,
+    }),
+    [mimeType, responseType],
+  )
+
   const query = useQuery({
+    enabled: isMediaTypeSupported,
     staleTime: Infinity,
-    queryKey: ['native-asset-img', policy, name, `${width}x${height}`, resizeMode],
+    queryKey: ['native-asset-img', policy, name, `${pWidth}x${pHeight}`, contentFit],
```

Comment:
Technically true, but we don't mix responseType, so it doesn't really change.
We're either using base64 or raw, not both

Reply:
If we remove the param then all of the support for base64 needs to be removed as well. Which very well could happen, it just needs to be decided first. Once the raw has been fully tested we can do that. There's still multiple tickets with cdn improvements.

Reply:
remember that `Infinity` here means, user session, since we are not persisting the react-query cache.

Reply:
yet, not sure about `expo-image`

Reply:
I added `expo-image` to improve on the base images, but I had it working without. Just with the same cache issue, because the image just doesn't reload for the same url.

Reply:
> remember that Infinity here means, user session, since we are not persisting the react-query cache.

My issue is not with the react-query cache. 
It's with the Image component. Neither Image from react-native nor Image from expo-image redownload an image with the same url

Reply:
we need `ETags` on next implementation